satellite/metabase/rangedloop/observerlivecount.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package rangedloop
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/spacemonkeygo/monkit/v3"
+
+	"storj.io/storj/satellite/metabase/segmentloop"
+)
+
+// LiveCountObserver reports a count of segments during loop execution.
+// This can be used to report the rate and progress of the loop.
+// Implements Observer, Partial and monkit.StatSource.
```

Comment:
A.H. says: It would be more robust and self-documenting to assert conformance to these interfaces at compile-time (see https://github.com/storj/storj/wiki/Style#implementing-interfaces).
var _ monkit.StatSource = (*LiveCountObserver)(nil)

Reply:
Done, I'll also check the other PR's.